### PR TITLE
Shank 495 benefits pension support

### PR DESCRIPTION
**Pull request by jhulbert-va**
https://vasdvp.atlassian.net/browse/SHANK-495

@jblefkove-va let me know if this is the correct way to do this. I grepped through the arcgis response and this was the only place I found the word "Pension"

### CODE REVIEW

**Comment by jblefkove-va** [comment]
@jhulbert-va This seems reasonable to me. Hitting the ArcGIS benefits endpoint today, I see these five facilities with pensions:

```
        "Facility_Name": "Philadelphia Regional Benefit Office",
        "Facility_Number": "310",
        "Other_Services": "Pension Claim Assistance, In-Service Death Claims",
--
        "Facility_Name": "Pittsburgh Regional Benefit Office",
        "Facility_Number": "311",
        "Other_Services": "Transition assistance is offered at Germany Benefits Delivery at Discharge (BDD) sites in Landstuhl and Korea BDD in Yongsan and Camp Humphreys. Homeless Assistance is available in filing compensation & pension claims. Additional services for homeless assistance will be coordinated through the Veterans Health Administration (VHA). ",
--
        "Facility_Name": "Reno Regional Benefit Office",
        "Facility_Number": "354",
        "Other_Services": "Outreach, Benefits Review, Compensation and Pension Claims Processing",
--
        "Facility_Name": "VBA Pecos Veterans Assistance Office",
        "Facility_Number": "354a",
        "Other_Services": "Outreach, Benefits Review, Compensation and Pension Claims Processing",
--
        "Facility_Name": "Manila Regional Benefit Office",
        "Facility_Number": "358",
        "Other_Services": "Filipino Veterans Equity Compensation assistance, Pension Benefits assistance, Survivors Benefits assistance, Outreach",
```

Can you check the values for these in the Rails API and see which ones they flag for pensions? (311 seems particularly questionable, since that is free-form text vs. the comma-delimited lists in the others)

**Comment by jhulbert-va** [comment]
> @jhulbert-va This seems reasonable to me. Hitting the ArcGIS benefits endpoint today, I see these five facilities with pensions:
> 
> ```
>         "Facility_Name": "Philadelphia Regional Benefit Office",
>         "Facility_Number": "310",
>         "Other_Services": "Pension Claim Assistance, In-Service Death Claims",
> --
>         "Facility_Name": "Pittsburgh Regional Benefit Office",
>         "Facility_Number": "311",
>         "Other_Services": "Transition assistance is offered at Germany Benefits Delivery at Discharge (BDD) sites in Landstuhl and Korea BDD in Yongsan and Camp Humphreys. Homeless Assistance is available in filing compensation & pension claims. Additional services for homeless assistance will be coordinated through the Veterans Health Administration (VHA). ",
> --
>         "Facility_Name": "Reno Regional Benefit Office",
>         "Facility_Number": "354",
>         "Other_Services": "Outreach, Benefits Review, Compensation and Pension Claims Processing",
> --
>         "Facility_Name": "VBA Pecos Veterans Assistance Office",
>         "Facility_Number": "354a",
>         "Other_Services": "Outreach, Benefits Review, Compensation and Pension Claims Processing",
> --
>         "Facility_Name": "Manila Regional Benefit Office",
>         "Facility_Number": "358",
>         "Other_Services": "Filipino Veterans Equity Compensation assistance, Pension Benefits assistance, Survivors Benefits assistance, Outreach",
> ```
> 
> Can you check the values for these in the Rails API and see which ones they flag for pensions? (311 seems particularly questionable, since that is free-form text vs. the comma-delimited lists in the others)

These are what I found:
```
vba_310
vba_330
vba_335
```

Doesn't look quite right when compared to what you sent though...

**Comment by jblefkove-va** [comment]
There's nothing in the source data for 330 or 335 to indicate that they offer pensions. So that may be stale data in the Rails API or some other error on their end. If we only want to match 310, you could match on `Pension Claim Assistance`.

**Comment by jhulbert-va** [comment]
> There's nothing in the source data for 330 or 335 to indicate that they offer pensions. So that may be stale data in the Rails API or some other error on their end. If we only want to match 310, you could match on `Pension Claim Assistance`.

If it is stale data, then it seems to me we'd want to list all of the ones we see in ArcGIS. That being said, you were worried about 311. Reading through it, I think that it should include Pensions in its list of benefits services.

File: facilities-collector/src/main/java/gov/va/api/lighthouse/facilitiescollector/BenefitsTransformer.java [hunk 1]
```diff
@@ -97,6 +98,10 @@
       benefitsServices.add(
           Facility.BenefitsService.VocationalRehabilitationAndEmploymentAssistance);
     }
+    if (attributes.otherServices() != null
+        && StringUtils.containsIgnoreCase(attributes.otherServices(), "PENSION")) {
```
Comment (jblefkove-va) [inline facilities-collector/src/main/java/gov/va/api/lighthouse/facilitiescollector/BenefitsTransformer.java:102]: You can remove the null check on `attributes.otherServices()`, `containsIgnoreCase` can handle it

**Pull request merged**